blockchain/block/testing.go

```diff
@@ -72,11 +74,17 @@ func (b *TestingBuilder) SetReceipts(receipts []*action.Receipt) *TestingBuilder
 
 // SignAndBuild signs and then builds a block.
 func (b *TestingBuilder) SignAndBuild(signerPrvKey crypto.PrivateKey) (Block, error) {
-	b.blk.Header.txRoot = b.blk.CalculateTxRoot()
+	var err error
+	b.blk.Header.txRoot, err = b.blk.CalculateTxRoot()
+	if err != nil {
+		log.L().Debug("error in getting hash", zap.Error(err))
+		return Block{}, errors.New("Failed to get hash")
+	}
 	b.blk.Header.pubkey = signerPrvKey.PublicKey()
 	h := b.blk.Header.HashHeaderCore()
 	sig, err := signerPrvKey.Sign(h[:])
 	if err != nil {
+		log.L().Debug("error in getting hash", zap.Error(err))
```

Comment:
"failed to sign block"

Reply:
done

Reply:
one final comment

Reply:
@dustinxie Please review now.

Reply:
> one final comment

@dustinxie All review comments incorporated.